packages/now-build-utils/src/detect-routes.ts

```diff
@@ -262,39 +299,14 @@ async function detectApiRoutes(
       };
     }
 
-    defaultRoutes.push(createRouteFromPath(file));
-  }
-
-  // 404 Route to disable directory listing
-  if (defaultRoutes.length > 0) {
-    if (featHandleMiss) {
-      defaultRoutes = [
-        { handle: 'miss' },
-        {
-          src: '/api/(.+)\\.\\w+',
-          dest: '/api/$1',
-          check: true,
-        },
-        {
-          status: 404,
-          src: '/api(/.*)?$',
-          continue: true,
-        },
-      ];
-    } else if (
-      defaultRoutes.some(
-        route =>
-          !isHandler(route) && route.dest && route.dest.startsWith('/api/')
-      )
-    ) {
-      defaultRoutes.push({
-        status: 404,
-        src: '/api(/.*)?$',
-      });
+    const out = createRouteFromPath(file, featHandleMiss, cleanUrls);
+    if (out.isDynamic) {
+      dynamicRoutes.push(out.route);
     }
+    defaultRoutes.push(out.route);
```

Comment:
Does this mean that all dynamic routes are in `dynamicRoutes` and in `defaultRoutes`?

Reply:
Correct. This is for backwards compatibility with feature flag off or on. In a future PR we could remove one of these.